packages/common/src/core/types/MaybeTranslate.ts

```diff
@@ -0,0 +1,12 @@
+/**
+ * Type that could be a string or an object with key property to be used for translation
+ */
+export type MaybeTranslate = string | { key: string };
```

Comment:
I ended up not going this route b/c the typings got messy, but I think this may be a useful construct, so I've left it in